src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -60,7 +73,35 @@ def create_configstore(client,
                                             enable_purge_protection=enable_purge_protection,
                                             create_mode=CreateMode.DEFAULT)
 
-    return client.begin_create(resource_group_name, name, configstore_params)
+    progress = IndeterminateStandardOut()
+
+    progress.write({"message": "Starting"})
+    config_store = client.begin_create(resource_group_name, name, configstore_params)
+
+    # # Poll request and create replica after store is created
+    while config_store.status() == ProvisioningStatus.RUNNING:
+        progress.spinner.step(label="Creating store")
+        config_store.wait(3)
+
+    if config_store.status() == ProvisioningStatus.SUCCEEDED and replica_name is not None:
```

Comment:
What happens when store creation fails? Are we guaranteed an error will be thrown?

Reply:
Yeah it still throws an error. wait() function can throw error